physlr/physlr.py

```diff
@@ -1189,106 +1189,109 @@ def assign_read_molecule(mxs, moltomxs, mol_counts, bx):
         return "_" + str(random.choice(max_hits)), 0, 1
 
     @staticmethod
-    def determine_molecules_biconnected_components(g, u):
-        "Separate bi-connected components."
-        cut_vertices = set(nx.articulation_points(g.subgraph(g.neighbors(u))))
-        components = list(nx.connected_components(g.subgraph(set(g.neighbors(u)) - cut_vertices)))
+    def detect_communities_biconnected_components(g, node_set):
+        """Separate bi-connected components. Return components."""
+        cut_vertices = set(nx.articulation_points(g.subgraph(node_set)))
+        components = list(nx.connected_components(g.subgraph(node_set - cut_vertices)))
         components.sort(key=len, reverse=True)
```

Comment:
Each `detect_communities_*` method needs to sort the communities by size.

Reply:
I suggest instead sorting just once in `determine_molecules` and removing all the other calls to `components.sort(key=len, reverse=True)`.

Reply:
I am wondering about the purpose of using `components.sort(key=len, reverse=True)`. If it is not needed before splitting, then I can simply move it to the line just before the return line of `determine_molecules` function. Do you confirm?

Reply:
Yes, I agree. The purpose of sorting is so that the scaffold backbones output by `physlr backbone` are sorted by size.